diff --git a/drivers/net/bnx2x/bnx2x_ethdev.c b/drivers/net/bnx2x/bnx2x_ethdev.c
index 6d5e982..38c2183 100644
--- a/drivers/net/bnx2x/bnx2x_ethdev.c
+++ b/drivers/net/bnx2x/bnx2x_ethdev.c
@@ -181,6 +181,8 @@ bnx2x_dev_configure(struct rte_eth_dev *dev)
 	PMD_INIT_FUNC_TRACE(sc);
 
 	sc->mtu = dev->data->dev_conf.rxmode.mtu;
+	sc->udp_rss = !!(dev->data->dev_conf.rx_adv_conf.rss_conf.rss_hf &
+			 RTE_ETH_RSS_NONFRAG_IPV4_UDP);
 
 	if (dev->data->nb_tx_queues > dev->data->nb_rx_queues) {
 		PMD_DRV_LOG(ERR, sc, "The number of TX queues is greater than number of RX queues");
@@ -682,9 +684,6 @@ bnx2x_common_dev_init(struct rte_eth_dev *eth_dev, int is_vf)
 	bnx2x_load_firmware(sc);
 	assert(sc->firmware);
 
-	if (eth_dev->data->dev_conf.rx_adv_conf.rss_conf.rss_hf & RTE_ETH_RSS_NONFRAG_IPV4_UDP)
-		sc->udp_rss = 1;
-
 	sc->rx_budget = BNX2X_RX_BUDGET;
 	sc->hc_rx_ticks = BNX2X_RX_TICKS;
 	sc->hc_tx_ticks = BNX2X_TX_TICKS;
